### DISCUSSION TITLE
Binary input does not respond after triggering

### DISCUSSION THREAD

**Discussion by yuantao313** [RavenSystem/esp-homekit-devices#2107] [Q&A] [answered]
Hello everyone I am building a multiple function switch with ESP-12F and this firmware.
I connected a relay to GPIO 12 and a button between GPIO 13 and VCC.
I constructed MELPHAA script like this:
```json
{
  "c": {
    "io": [
      [[12], 2],
      [[13], 6]
    ],
    "l": 2,
    "b": [[0, 5]]
  },
  "a": [
    {
      "0": { "r": [[12, 0]] },
      "1": { "r": [[12, 1]] },
      "b": [[13]],
      "t": 1,
      "s": 5
    }
  ]
}
```
Then I deployed the software and hardware. A problem occured: when I tried to switch the status of the relay by the button, it would work once and stop working for a while, then it can work again but lost respond after triggering. It seems that it has a "cooldown time".
Did anyone encounter this problem? I really need your help!

**Comment by Tellicious** [comment, marked as answer]
According to your description and the script, there seems to be nothing wrong in this behavior. If the button is between GPIO13 and VCC, you would need to have a pull-down resistor to allow the voltage to drop and the input to reset.

However ESP8266 (the micro inside ESP-12F) doesn't feature internal pull-down resistors, so you have 2 options:
1. install a pull-down resistor (4.7kOhm would work) between GPIO13 and GND and change its initialization to `[[13],6,0,1]` so to enable the invert GPIO behavior (active high)
2. connect the button between GPIO13 and GND and change its initialization to `[[13],6,1]` so to enable the internal pull-up

**Reply by mmmjsz** [reply]
The fisrt option works for me. Thanks a lot.

**Reply by yuantao313** [reply]
Though I realized it seveal days after I posted the discussion, I still appreciate your help. It is needful to check the document of the chip. 😀
